File: OpenCV/cppTest/mouse.cpp
#include "opencv2/opencv.hpp"
#include "opencv2/freetype.hpp"
#include <iostream>
#include <string>

using namespace cv;
using namespace std;

String folder = "/home/test/Desktop/AIOT_2024_ROBOT/OpenCV/cppTest/data/";

void onMouse(int event, int x, int y, int flags, void *data);

int main()
{
    Mat src = imread(folder + "lenna.bmp", IMREAD_COLOR);
    namedWindow("img");
    setMouseCallback("img", onMouse, ((void*)&src));

    imshow("img", src);
    waitKey(0);

    return 0;
}

void onMouse(int event, int x, int y, int flags, void *data)
{
    Mat *src = (Mat *)data;
    Mat temp;
    temp = (*src).clone();
    switch (event)
    {
    case EVENT_LBUTTONDOWN:
        cout << x << y << endl;
        circle(*src, Point(x, y), 10, Scalar(255, 0, 0), -1);
        imshow("img", *src);
        break;
    case EVENT_MOUSEMOVE:
        rectangle(temp, Rect(x, y, 50, 50), Scalar(0, 0, 255), 5);
        imshow("img", temp);
        break;

    default:
        break;
    }
}

File: OpenCV/cppTest/pythonTest/mouse.py
import cv2
import numpy as np

def on_mouse(event, x, y, flags, param):
    src = param['src']
    if event == cv2.EVENT_LBUTTONDOWN:
        print(x, y)
        cv2.circle(src, (x, y), 10, (255, 0, 0), -1)
        cv2.imshow("img", src)

def main():
    folder = "/home/test/Desktop/AIOT_2024_ROBOT/OpenCV/cppTest/data/"
    src = cv2.imread(folder + "lenna.bmp", cv2.IMREAD_COLOR)

    param = {'src':src}
    cv2.namedWindow("img")
    cv2.setMouseCallback("img", on_mouse, param)

    
    cv2.imshow("img", src)
    cv2.waitKey(0)
        

    
if __name__ == "__main__":
    main()